apis/python/src/tiledbsoma/uns_group.py

```diff
@@ -145,71 +213,23 @@ def from_anndata_uns(self, uns: Mapping[str, Any]) -> None:
 
+        futures = []
+        max_workers = self._soma_options.max_thread_pool_workers
+
+        threadsafe_args = {}
         for key in uns.keys():
-            component_uri = child_uris[key]
-            value = uns[key]
-
-            if key == "rank_genes_groups":
-                # TODO:
-                # This is of type 'structured array':
-                # https://numpy.org/doc/stable/user/basics.rec.html
-                #
-                # >>> a.uns['rank_genes_groups']['names'].dtype
-                # dtype([('0', 'O'), ('1', 'O'), ('2', 'O'), ('3', 'O'), ('4', 'O'), ('5', 'O'), ('6', 'O'), ('7', 'O')])
-                # >>> type(a.uns['rank_genes_groups']['names'])
-                # <class 'numpy.ndarray'>
-                #
-                # We don’t have a way to model this directly in TileDB schema right now. We support
-                # multiplicities of a single scalar type, e.g. a record array with cell_val_num==3
-                # and float32 slots (which would correspond to numpy record array
-                # np.dtype([("field1", "f4"), ("field2", "f4"), ("field3", "f4",)])). We don’t
-                # support nested cells, AKA "list" type.
-                #
-                # This could, however, be converted to a dataframe and ingested that way.
-                log_io(
-                    None, f"{self._indent}Skipping structured array: {component_uri}"
-                )
-                continue
-
-            if isinstance(value, Mapping):
-                # Nested data, e.g. a.uns['draw-graph']['params']['layout']
-                subgroup = UnsGroup(uri=component_uri, name=key, parent=self)
-                subgroup.from_anndata_uns(value)
-                self._add_object(subgroup)
-                continue
-
-            # Write scalars as metadata, not length-1 component arrays
-            if isinstance(value, np.str_):
-                with self._open("w") as G:
-                    G.meta[key] = value
-                # TODO: WUT
-                # Needs explicit cast from numpy.str_ to str for tiledb.from_numpy
-                continue
-
-            if isinstance(value, (int, float, str)):
-                # Nominally this is unit-test data
-                with self._open("w") as G:
-                    G.meta[key] = value
-                continue
-
-            # Everything else is a component array, or unhandleable
-            array = UnsArray(uri=component_uri, name=key, parent=self)
-
-            if isinstance(value, pd.DataFrame):
-                array.from_pandas_dataframe(value)
-                self._add_object(array)
-
-            elif isinstance(value, sp.csr_matrix):
-                array.from_scipy_csr(value)
-                self._add_object(array)
-
-            elif array._maybe_from_numpyable_object(value):
-                self._add_object(array)
+            threadsafe_args[key] = (key, child_uris[key], uns[key])
```

Comment:
I don't follow why we're doing this here; it seems like you could do the same thing with `executor.submit` without building this dict first.